src/providers/WorkflowCore.LockProviders.MySQL/MySqlLockProvider.cs

```diff
@@ -0,0 +1,138 @@
+﻿using System;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Logging;
+using WorkflowCore.Interface;
+using System.Data;
+using System.Collections.Generic;
+using System.Threading;
+using MySql.Data.MySqlClient;
+
+namespace WorkflowCore.LockProviders.MySQL
+{
+    public class MySqlLockProvider : IDistributedLockProvider
+    {
+        private const string Prefix = "wfc";
+
+        private readonly string _connectionString;
+        private readonly ILogger _logger;
+        private readonly Dictionary<string, MySqlConnection> _locks = new Dictionary<string, MySqlConnection>();
+        private readonly AutoResetEvent _mutex = new AutoResetEvent(true);
+
+        public MySqlLockProvider(string connectionString, ILoggerFactory logFactory)
+        {
+            _logger = logFactory.CreateLogger<MySqlLockProvider>();
+            var csb = new MySqlConnectionStringBuilder(connectionString);
+            csb.Pooling = true;
+            _connectionString = csb.ToString();
+        }
+
+
+        public async Task<bool> AcquireLock(string Id, CancellationToken cancellationToken)
+        {
+            if (_mutex.WaitOne())
+            {
+                try
+                {
+                    var connection = new MySqlConnection(_connectionString);
```

Comment:
Does this come from a connection pool or is it establishing a fresh connection everytime?

Reply:
It comes from a pool. You can see the code to set Pooling = true for connection string in the constructor.